vms/platformvm/state/diff.go

```diff
@@ -403,6 +406,38 @@ func (d *diff) AddTx(tx *txs.Tx, status status.Status) {
 	}
 }
 
+func (d *diff) GetRewardConfig(subnetID ids.ID) (reward.Config, error) {
+	parentState, ok := d.stateVersions.GetState(d.parentID)
+	if !ok {
+		return reward.Config{}, fmt.Errorf("%w: %s", ErrMissingParentState, d.parentID)
+	}
+
+	primaryNetworkCfg, err := parentState.GetRewardConfig(subnetID)
```

Comment:
reward section of config should not really change. But I feel it's better to pull the config up from state.State than injecting the config into state.Diff via its constructor. We avoid potential errors of injecting different configs for different diffs.